File: 7kyu/DnaStrand.java
/*
 * Complementary DNA kata - Java
 *
 * @author Unai de la O
 */

public class DnaStrand {
  public static String makeComplement(String dna) {
    String finalStr = "";

    for (int i = 0; i < dna.length(); i++) {
      char actual = dna.charAt(i);

      if (actual == 'A') {
        finalStr += 'T';
      }
      else if (actual == 'T') {
        finalStr += 'A';
      }
      else if (actual == 'C') {
        finalStr += 'G';
      }
      else if (actual == 'G') {
        finalStr += 'C';
      }
    }
    return finalStr;
  }

 public static void main(String[] args) {
   String test = "";

   System.out.println(makeComplement(test));
 }
}

File: 7kyu/DnaStrand.cs
// Complementary DNA kata - C#
//
// Author: Unai de la O

using System;

public class DnaStrand
{
    public static string MakeComplement(string dna)
    {
        string finalStr = "";

        for (int i = 0; i < dna.Length; i++)
        {
          char actual = dna[i];

          if (actual == 'A')
            finalStr += 'T';
          else if (actual == 'T')
            finalStr += 'A';
          else if (actual == 'C')
            finalStr += 'G';
          else if (actual == 'G')
            finalStr += 'C';
        }
        return finalStr;
    }

    public static void Main(string[] args)
    {
      // string vacío, necesario para que se pudiesen pasar todos los tests
      string test = "";

      Console.WriteLine(MakeComplement(test));
    }
}
